From 20cbffed79338aac4cc3ecd90fe9f50fc7cd3086 Mon Sep 17 00:00:00 2001
From: Kevin O'Connor <kevin@koconnor.net>
Date: Sat, 9 Jul 2011 14:49:13 -0400
Subject: [PATCH 58/91] Remove now unneeded find_pci().

Signed-off-by: Kevin O'Connor <kevin@koconnor.net>
---
 src/boot.c |   13 -------------
 1 files changed, 0 insertions(+), 13 deletions(-)

diff --git a/src/boot.c b/src/boot.c
index 4e86477..119f290 100644
--- a/src/boot.c
+++ b/src/boot.c
@@ -98,22 +98,9 @@ find_prio(const char *glob)
 
 #define FW_PCI_DOMAIN "/pci@i0cf8"
 
-static struct pci_device *
-find_pci(u16 bdf)
-{
-    struct pci_device *pci;
-    foreachpci(pci) {
-        if (pci->bdf == bdf)
-            return pci;
-    }
-    return NULL;
-}
-
 static char *
 build_pci_path(char *buf, int max, const char *devname, struct pci_device *pci)
 {
-    if (!pci)
-        return buf;
     // Build the string path of a bdf - for example: /pci@i0cf8/isa@1,2
     char *p = buf;
     if (pci->parent) {
-- 
1.7.5.4

