From 871c92991b65fff584d8200c767f8e1a08f8c4a1 Mon Sep 17 00:00:00 2001
From: Kevin O'Connor <kevin@koconnor.net>
Date: Tue, 12 Jul 2011 21:14:10 -0400
Subject: [PATCH 65/91] Updates to TODO list.

---
 TODO |    8 --------
 1 files changed, 0 insertions(+), 8 deletions(-)

diff --git a/TODO b/TODO
index d49aeb8..23f26c0 100644
--- a/TODO
+++ b/TODO
@@ -1,5 +1,3 @@
-If POST is rerun, try to do a machine reboot.
-
 Review changes committed to coreboot, virtualbox, qemu, kvm, and bochs
 cvs tip.
   * bochs cvs (20100104):
@@ -12,9 +10,6 @@ cvs tip.
     -- lots of mouse changes, logo, scsi/etherboot hooks,
        floppy data rate?, int19 calls post
 
-Possibly move code from entry_post in romlayout.S to C code in
-handle_resume and always call 16bit C code on post.
-
 The __call16 code does a long jump to the interrupt trampolines - this
 is unnecessary.
 
@@ -23,7 +18,4 @@ extensions?
 
 Possibly add option to eliminate tsc based delays on emulators.
 
-Add a kconfig style configuration program instead of requiring users
-to modify config.h.
-
 Possibly support sending debug information over EHCI debug port.
-- 
1.7.5.4

